source_modelling/sources.py

```diff
@@ -422,6 +464,66 @@ def wgs_depth_coordinates_to_fault_coordinates(
             raise ValueError("Specified coordinates do not lie in plane")
         return np.clip(fault_local_coordinates, 0, 1)
 
+    def rrup_distance(self, point: np.ndarray) -> float:
+        """Compute RRup Distance between a fault and a point.
+
+        Parameters
+        ----------
+        point : np.ndarray
+            The point to compute distance to (in lat, lon, depth format)
+
+        Returns
+        -------
+        float
+            The rrup distance (in metres) between the point and the fault geometry.
+
+        Raises
+        ------
+        ValueError
+            If the optimisation routine fails to converge for the point.
+        """
+        point_nztm = coordinates.wgs_depth_to_nztm(point)
+
+        def fault_coordinate_distance(fault_coordinates: np.ndarray) -> float:
+            fault_point = coordinates.wgs_depth_to_nztm(
+                self.fault_coordinates_to_wgs_depth_coordinates(fault_coordinates)
+            )
+            return point_nztm - fault_point
+
+        res = sp.optimize.least_squares(
```

Comment:
I think a non-optimization based approach would be more suitable. Would make it deterministic (which is what the rrup calculation is), and also easier to scale in the future in the case of many points.